From a99ddf264b4d14570fa4f510893c56cae9975d2d Mon Sep 17 00:00:00 2001
From: McKnight22 <tao.wang.2261@gmail.com>
Date: Mon, 15 May 2023 18:05:06 +0800
Subject: [PATCH] [riscv64][android] Fix identifier libs undefined error

Identifier "libs" is not declared at the top of component "v8_libbase"
as other identifiers eg. "data". Meanwhile, for android case, "libs"
definition is introduced only when "current_toolchain==host_toolchain".
So, it causes the error when generating build configuration for riscv64
android.

co-authors: McKnight22 <tao.wang.2261@gmail.com>
            Wang Chen <wangchen20@iscas.ac.cn>

Bug: v8:13998
Change-Id: I2b7b705b4bd980c7213051b1c49515b044300a5f
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4531044
Commit-Queue: Yahan Lu <yahan@iscas.ac.cn>
Reviewed-by: Michael Achenbach <machenbach@chromium.org>
Cr-Commit-Position: refs/heads/main@{#87712}
---
 AUTHORS  | 2 ++
 BUILD.gn | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/AUTHORS b/AUTHORS
index 27b2c27de6b..2b393ad3876 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -252,6 +252,7 @@ Sylvestre Ledru <sledru@mozilla.com>
 Takeshi Yoneda <takeshi@tetrate.io>
 Taketoshi Aono <brn@b6n.ch>
 Tao Liqiang <taolq@outlook.com>
+Tao Wang <tao.wang.2261@gmail.com>
 Teddy Katz <teddy.katz@gmail.com>
 Thomas Young <wenzhang5800@gmail.com>
 Tiancheng "Timothy" Gu <timothygu99@gmail.com>
@@ -268,6 +269,7 @@ Vlad Burlik <vladbph@gmail.com>
 Vladimir Krivosheev <develar@gmail.com>
 Vladimir Shutoff <vovan@shutoff.ru>
 Wael Almattar <waelsy123@gmail.com>
+Wang Chen <wangchen20@iscas.ac.cn>
 WANG Xuerui <git@xen0n.name>
 Wei Wu <lazyparser@gmail.com>
 Wenlu Wang <kingwenlu@gmail.com>
diff --git a/BUILD.gn b/BUILD.gn
index f4fda5d3606..e9aa3775b2f 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -5846,6 +5846,8 @@ v8_component("v8_libbase") {
 
   deps = [ ":v8_config_headers" ]
 
+  libs = []
+
   data = []
 
   data_deps = []
-- 
2.35.1

